app/migration/parity_check/request_builder.rb

```diff
@@ -115,5 +115,10 @@ def pagination_enabled?
     def dynamic_request_content
       @dynamic_request_content ||= DynamicRequestContent.new(lead_provider:)
     end
+
+    def dynamic_request_contents(value)
+      @dynamic_request_contents ||= {}
```

Comment:
We're not testing this anywhere I don't think 🤔 I think it will be hard to test as well; would it be cleaner/easier to test if we just memoize the methods in `DynamicRequestContent` so it consistently returns the same values for a given lead provider/instance?

FWIW I'm adding some request consistency checks to the `Client` that should cover it more generally as part of the PR to expose the request body/URIs in the interface: https://github.com/DFE-Digital/register-early-career-teachers-public/commit/e4d6c3f50aae43b992344f4743a26dbace6b33d6

Reply:
Hi @ethax-ross makes sense.. I've moved the memoization to `DynamicRequestContent` instead. Added a spec to test it. I hope it's all good. 👍